consensus/obcpbft/viewchange.go

```diff
@@ -435,7 +435,7 @@ func (instance *pbftCore) processNewView2(nv *NewView) error {
 
 	logger.Debug("Replica %d done cleaning view change artifacts, calling into consumer", instance.id)
 
-	instance.manager.inject(viewChangedEvent{})
+	instance.sendViewChangedEvent = true
```

Comment:
Ultimately, I would see the view change path switching to be driven off events.  There would be something like a `viewChangeQuorumEvent` which would be returned from the the `recvViewChange`, which would trigger the `processNewView` path (as would receiving a new view).  Then, the `processNewView` path should return the `viewChangedEvent`.